### DISCUSSION TITLE
Regarding course confusion

### DISCUSSION THREAD

**Discussion by Tanush103** [snigi-gupta/AttendingUniversityAtBuffalo#23] [General]
Hi Snigdha ,
Thanks for creating this discussion forum and replying to people's queries .It feels great to have a active alumni replying to the queries as I know its difficult to get time between work for this.

I was going through the courses and finalized some of them. For some courses I have seen your github page where various student have listed opinions about the course .It will be really helpful if you can help me please. I am interested in Big Data and Data Engineering Domain and would be interested in doing specializations in that area , so I have selected subjects according to that .

1.CSE 529 Algorithms for Modern Computing Systems OR CSE 531 Analysis of Algorithms-(What is the difference between the two courses. Also since I am from a non CS Background, is there any prerequisite related with any one of these? Like student must have CS Undergrad?)

2.CSE 535 Information Retrieval,
CSE 562 Database Systems,
CSE 521 Introduction to Operating Systems,
CSE 587 Data-Intensive Computing(I have read good reviews about information retrieval on your github page. Do you have any idea about how good is CSE587 Course? For Data Engineering and Analytics domain which two courses out of above four would you suggest?

3.
CSE 635 Advanced Topics in Information Retrieval
CSE 636 Data Integration(Any reviews on these courses ???Like they are 600 level courses so which would be better (I am inclined more towards CSE 636 as more related to data but for that CSE 562 is a prerequisite)

4.For the electives I have shortlisted two courses 
MGS 653 Social Network Analytics
MGS 655 Distributed Computing and Big Data(Any idea how are these courses)

Any more courses you can suggest for Big Data or Data Engineering would be really be very helpful.
Sorry for this long query .But Actually I'm confused which courses to consider.

Thanks for continuing support and  helping students always.

**Comment by snigi-gupta** [comment]
@Tanush103,

1. 529 is about algorithms on parallel systems. It slightly covers DAC, but is more about parallel computing and you can check [professor's link](https://cse.buffalo.edu/faculty/miller/). The professor has good reviews. 531 is core CS fundamental basically DSA(Data Structure and Algorithms). 
2. 535 has a good balance of data-engineering and information-retrieval concepts. You would do data-crawling, data pre-processing and also perform analysis on the data. This project is on my resume. With 587 you would work on random-forests and decision trees. I didn't pursue this course, so I would suggest you copy the link of this discussion page and paste it on LinkedIn in chate with other UB graduated students to share their views. That way you will get multiple view-points. 562 and 521 are both different courses. Database is about building a distributed database. It is a hard course. Operating Systems is about kernals and OS as the name suggest. This too is a hard course. Both these course do not have a big data / data engineering component.
3. 635 is basically Natural Language Processing (it has been renamed). I enjoyed this course as I built a machine learning system. I learnt about Siamese Networks and BART technologies. There is a sem-long project component. This project is on my resume. I am not aware about 636, but if it interests you definitely give it a try! again, you can use this discussion link to get your answers from someone who would have taken this subject. Probably share in a whatsapp group? 🤔 
4. You need to get permission to take course outside of department. If they are relevant to you degree, then it shouldn't be a problem. I haven't taken these courses, but I would suggest the best bet is to a) audit the course when you have joined the school. b) email prof now, and maybe the prof can send you a guideline. Also check prof's website.
5. For big data 587 is definitely a course to take. [checkout this link](https://catalog.buffalo.edu/academicprograms/data-intensive_computing_cert.html), might give you some idea.

**Comment by Tanush103** [comment]
Thanks a lot Snigdha...That was quite helpful..😊
